clang/lib/ASTMatchers/Dynamic/Registry.cpp

```diff
@@ -319,6 +319,7 @@ RegistryMaps::RegistryMaps() {
   REGISTER_MATCHER(hasExplicitSpecifier);
   REGISTER_MATCHER(hasExternalFormalLinkage);
   REGISTER_MATCHER(hasFalseExpression);
+  REGISTER_MATCHER(hasFoldInit);
```

Comment:
I think that you should also register here other matchers like isBinaryFold to make them visible in clang-query.

Reply:
done